### PR TITLE
Add matomo_tag to /register pages

### LINKED ISSUE #9534: Matomo tracking code missing from /register page
## Description

Matomo tracking code seems to be missing from the `/register` page, although it can be found in other pages.

## Expected Behavior

Matomo tracking code should be found in all pages (right?)

## Actual Behaviour

Matomo tracking code missing from `/register` page.

## Steps to Reproduce

1. Log in as a customer.
2. Visit `view-source:https://openfoodnetwork.ca/register`
3. Notice the missing tracking code.

## Workaround

In principle, this completely blocks tracking on the /register pages... No workaround in that case. Not sure though.

## Severity

bug-s3

## Your Environment

* Version used: v4.2.4
* Browser name and version: Firefox 102
* Operating System and version (desktop or mobile): Ubuntu 22.04 Jammy :sparkles:

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Closes #9534
Prepares https://github.com/openfoodfoundation/wishlist/issues/257

Currently, we are adding both the tracking code and the code manager to our page layouts, because these were defined in the same file.

This PR:
- separates the respective matomo code snippets into two files:
  - `matomo_tag` (should only contain tag manager code)
  - `matomo_tracking` (should only contain general tracking code)

- adds the matomo_tag to the `/register` page - complying with #https://github.com/openfoodfoundation/wishlist/issues/257
- adds both tracking codes to our darkswarm template -> this should keep the same tracking code on all pages **as before**. #https://github.com/openfoodfoundation/wishlist/issues/257 is not closed by this issue.

#### What should we test?

Best tested on staging-FR.

A) We'll need to check the presence/absence of the matomo code snippets **A and B**:

image

**Before staging**

Test 1

- Login as a customer
- Visit `/register` page
- check the source code of the URL
- Expected outcome: there should be no matomo code on this page **(no code)**

Test 2

- Login as a customer
- Visit the homepage
- check the source code of the URL
- Expected outcome: the tag manager code and the tracking should be visible **(A + B)**

**After staging**

Test 1

- Login as a customer
- Visit `/register` page
- check the source code of the URL
- Expected outcome: the tag manager code should be visible **(A)**

Test 2

- Login as a customer
- Visit the homepage
- check the source code of the URL
- Expected outcome: the tag manager code and the tracking should be visible -> i.e., no change is expected **(A + B)**

B) In addition to this, we should run a sanity test to make sure we have no regressions on our Matomo implementation. The behavior should not change with this PR, i.e. before and after staging we should be able to:

- trigger tags -> to test, user the URL `https://staging.coopcircuits.fr/?mtmPreviewMode=aSBgrfIY#/login` (make sure cookies are accepted)
- track site visits (site_id=15, for staging FR), as before (make sure cookies are accepted)

#### Release notes

Changelog Category: Technical changes

Adds matomo_tag to /register pages template

### CODE REVIEW

**Review by jibees — APPROVED** [review]
Looks good to me! ;)

**Comment by filipefurtad0** [comment]
I think this PR is outdated. I don't think there is much sense in merging following this advice from Matomo support.

My proposal would be to change this PR and:
- create a new file containing only the tag manager matomo code snippet
- add this new tag for now _only_ to the `/register` page

File: app/views/layouts/registration.html.haml [hunk 1]
```diff
@@ -10,6 +10,7 @@
       = favicon_link_tag "/favicon-staging.ico"
     %link{href: "https://fonts.googleapis.com/css?family=Roboto:400,300italic,400italic,300,700,700italic|Oswald:300,400,700", rel: "stylesheet", type: "text/css"}
     = render "layouts/matomo_tag"
+    = render "layouts/matomo_tracking"
```
Comment (jibees) [inline app/views/layouts/registration.html.haml:13]: The commit message indicates: 
`Adds the tracking matomo code to the darkswarm template`

But it seems that only `app/views/layouts/registration.html.haml` file did change on this commit. May I missed something?

**Review by jibees — COMMENTED** [review]
Just one question ;)

File: app/views/layouts/registration.html.haml (hunk 1 above)
Comment (filipefurtad0) [reply app/views/layouts/registration.html.haml:13]: Wrong template...Correcting.

**Review by mkllnk — APPROVED** [review]
Your code changes look like they would achieve what you described above but I have to admit that I don't really know how Matomo should be set up. I had a quick look at the references you gave in the issue but found that I would need to spend some more time reading.

This should solve the issue for now though.

**Comment by filipefurtad0** [comment]
Thanks for testing @drummer83 .

Some additional context to the test mentioned in B)

i) **trigger tags**
- visit the URL https://staging.coopcircuits.fr/?mtmPreviewMode=aSBgrfIY#/login (make sure cookies are accepted). It should display the homepage with a preview Matomo section on the bottom:

image

- Click "Login"; it should open the Login modal
- On the login modal, select the Sign-up tab
- Click "Sign-up now"
- A trigger event should be displayed on the Matomo section. It should have a green tick next to it.
- Click the triggered event on the Matomo section: the Fired Count should be at least 1:

image

- **track site visits (site_id=15, for staging FR)**
Accepting tracking cookies is a condition for this feature to work. One option is to use Chrome (allows tracking cookies by default) or change the settings of your Firefox browser for this test.

- Visit https://staging.coopcircuits.fr/; click around, visit shops, and/or login
- Log in to your Matomo dashboard - try to find your visit, within the data (country, device, time, etc).

Please note: it can be a bit challenging to find your visit because the data are not collected in real time. It is only after a few hours that your visit will appear. So it might give an initial impression that Matomo is not really tracking correctly. My suggestion is to wait a few hours between your actions on the website and the reading of results, on the Matomo platform.

Example:

I just visited the staging server's homepage with a IP from Portugal, Ubuntu + Chrome (direct entry). Right now, Matomo displays:

image

So it clearly is not tracking "me" yet. I'll check again later on.

**Comment by filipefurtad0** [comment]
Forgot to add: the post above could in principle serve as testing notes (tracking data pending, will update later), as this has been performed with the PR staged.

But I guess it would be great if you could reproduce this behavior though, it also could help in getting more familiar with Matomo. What do you think @drummer83 ?

**Comment by drummer83** [comment]
Thank you for the additional information, @filipefurtad0!

I can confirm that the tag is being fired! :+1: 
image

And I can also see that our visits from yesterday have been tracked (the PR was staged on 30th August)! :+1: 
image

So, if there is nothing else coming to your mind, we are good to merge here! :tada:

**Pull request merged**